lib/api/axiosFactory.js

```diff
@@ -0,0 +1,274 @@
+const { firmCredentials } = require("./firmCredentials");
+const axios = require("axios");
+const { consola } = require("consola");
+const pkg = require("../../package.json");
+
+class AxiosFactory {
+  constructor() {}
+
+  /**
+   * Create an axios instance for a given type and environment id.
+   * It will add the necessary headers and interceptors to handle the authorization, and refresh the tokens if needed.
+   * The created instance will be used to make requests to the Silverfin API.
+   * @param {String} type - The type of instance to create (firm or partner)
+   * @param {Number} envId - The environment id to create the instance for
+   * @returns {Object} - The created axios instance
+   */
+  static createInstance(type, envId) {
```

Comment:
Would it make sense to store these instances in memory iso creating a new one on every call? This would then become more of something like a `findOrCreateInstance` or `loadInstance`:

```
const key = `${type}_${envId}`
axios_instances[key] ||= this.#createAxiosInstanceForFirms(envId)
return axios_instances[key]
```